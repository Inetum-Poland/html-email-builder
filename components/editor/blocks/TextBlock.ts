import type { BlockProperties } from "grapesjs";
import { Text } from "lucide-static";

export const TextBlock = (): BlockProperties => {
  const { $i18n: { t } } = useNuxtApp();

  return {
    label: t("text"),
    media: Text,
    content: `<mj-text>${t("insertTextHere")}</mj-text>`,
    activate: true,
    category: t("genericBlocks"),
  };
};
