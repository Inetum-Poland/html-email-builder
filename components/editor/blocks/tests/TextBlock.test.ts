import { expect, test } from "vitest";
import { TextBlock } from "../TextBlock";

test("renders the Text block", () => {
  expect(TextBlock()).toMatchSnapshot();
});
