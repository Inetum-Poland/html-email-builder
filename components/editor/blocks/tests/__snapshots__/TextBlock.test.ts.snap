// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the Text block 1`] = `
{
  "activate": true,
  "category": "Generic blocks",
  "content": "<mj-text>Insert text here...</mj-text>",
  "label": "Text",
  "media": "
<svg
  class="lucide lucide-text"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M17 6.1H3" />
  <path d="M21 12.1H3" />
  <path d="M15.1 18H3" />
</svg>
",
}
`;
